os/net/ipv6/sicslowpan.c

```diff
@@ -1632,25 +1623,15 @@ output(const linkaddr_t *localdest)
     set_packet_attrs();
   }
 
-  /*
-   * The destination address will be tagged to each outbound
-   * packet. If the argument localdest is NULL, we are sending a
-   * broadcast packet.
-   */
-  if(localdest == NULL) {
-    linkaddr_copy(&dest, &linkaddr_null);
-  } else {
-    linkaddr_copy(&dest, localdest);
-  }
-
   LOG_INFO("output: sending IPv6 packet with len %d\n", uip_len);
 
   /* copy over the retransmission count from uipbuf attributes */
   packetbuf_set_attr(PACKETBUF_ATTR_MAX_MAC_TRANSMISSIONS,
                      uipbuf_get_attr(UIPBUF_ATTR_MAX_MAC_TRANSMISSIONS));
 
-/* Calculate NETSTACK_FRAMER's header length, that will be added in the NETSTACK_MAC */
-  packetbuf_set_addr(PACKETBUF_ADDR_RECEIVER, &dest);
+  /* Copy destination address to packetbuf */
+  packetbuf_set_addr(PACKETBUF_ADDR_RECEIVER, localdest ?: &linkaddr_null);
```

Comment:
This ternary operator shortcut is not part of standard C, as far as I know. It would be preferable to avoid compiler-specific extensions in the core modules.